### PR TITLE
Add test for regexp langlib usage in query actions and query expressions

### LINKED ISSUE #40505: [Bug]: Compilation failed when using ballerina/lang.regexp's isFullMatch method
### Description

Compilation failed when using ballerina/lang.regexp's isFullMatch method
```re `(\S+)\.(\S+)`.isFullMatch(logRecord.exception) == false```
image

And gives the following log record in the terminal

```ballerina: Oh no, something really went wrong. Bad. Sad.

We appreciate it if you can report the code that broke Ballerina in
https://github.com/ballerina-platform/ballerina-lang/issues with the
log you get below and your sample code.

We thank you for helping make us better.

[2023-05-24 23:13:25,957] SEVERE {b7a.log.crash} - null 
java.lang.AssertionError
        at org.wso2.ballerinalang.compiler.tree.BLangNodeVisitor.visit(BLangNodeVisitor.java:1072)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangRegExpTemplateLiteral.accept(BLangRegExpTemplateLiteral.java:36)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.acceptNode(QueryDesugar.java:2315)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:1561)
        at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:54)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:1541)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:50)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.acceptNode(QueryDesugar.java:2315)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.lambda$visit$13(QueryDesugar.java:2122)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:2122)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangBlockStmt.accept(BLangBlockStmt.java:65)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.acceptNode(QueryDesugar.java:2315)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:2166)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangIf.accept(BLangIf.java:60)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.acceptNode(QueryDesugar.java:2315)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.lambda$visit$13(QueryDesugar.java:2122)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:2122)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangBlockStmt.accept(BLangBlockStmt.java:65)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.acceptNode(QueryDesugar.java:2315)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.lambda$visit$13(QueryDesugar.java:2122)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:2122)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangBlockStmt.accept(BLangBlockStmt.java:65)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.acceptNode(QueryDesugar.java:2315)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:2165)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangIf.accept(BLangIf.java:60)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.lambda$visit$5(QueryDesugar.java:1531)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:1531)
        at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:60)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:1516)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:76)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.visit(QueryDesugar.java:1503)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangLambdaFunction.accept(BLangLambdaFunction.java:64)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.addDoFunction(QueryDesugar.java:852)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.buildStream(QueryDesugar.java:524)
        at org.wso2.ballerinalang.compiler.desugar.QueryDesugar.desugar(QueryDesugar.java:378)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:8508)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangQueryAction.accept(BLangQueryAction.java:47)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewriteExpr(Desugar.java:8926)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:1477)
        at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:54)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8901)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:1613)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:50)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8901)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8939)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewriteStmt(Desugar.java:8945)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:1380)
        at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:60)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8901)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:1426)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:76)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8901)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:830)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:167)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8901)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.perform(Desugar.java:473)
        at io.ballerina.projects.internal.CompilerPhaseRunner.desugar(CompilerPhaseRunner.java:216)
        at io.ballerina.projects.internal.CompilerPhaseRunner.performBirGenPhases(CompilerPhaseRunner.java:152)
        at io.ballerina.projects.ModuleContext.generateCodeInternal(ModuleContext.java:451)
        at io.ballerina.projects.ModuleCompilationState$4.generatePlatformSpecificCode(ModuleCompilationState.java:132)
        at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:385)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:170)
        at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:139)
        at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:117)
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1134)
        at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:170)
        at io.ballerina.projects.JBallerinaBackend.from(JBallerinaBackend.java:116)
        at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:165)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.RunCommand.execute(RunCommand.java:221)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:51)
 
ERROR [log_analyzer_service:(1:1,1:1)] Compilation failed due to an unhandled exception
error: compilation contains errors```

### Steps to Reproduce

Compilation failed when using ballerina/lang.regexp's isFullMatch method
```re `(\S+)\.(\S+)`.isFullMatch(logRecord.exception) == false```

### Affected Version(s)

Ballerina : "2201.4.1"

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Fixes #40505

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/query/query_action_or_expr.bal [hunk 1]
```diff
@@ -915,6 +915,28 @@ function testQueryActionWithQueryExpression() {
...
+    assertEquality(["a", "aab", "ac"], res);
+}
+
+function testQueryExprWithRegExpLangLibs() {
+    string[] res = from var item in ["a", "aab", "bc", "ac"] 
+    where re `a.*`.isFullMatch(item)
+    select item;
+    
+    assertEquality(["a", "aab", "ac"], res);
+}
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/query/query_action_or_expr.bal:938]: Add a test where a interpolated regex is being used within the query

Comment (LakshanWeerasinghe) [reply tests/jballerina-unit-test/src/test/resources/test-src/query/query_action_or_expr.bal:938]: Added.

File: tests/jballerina-unit-test/src/test/resources/test-src/query/query_action_or_expr.bal [hunk 2]
```diff
@@ -915,6 +915,41 @@ function testQueryActionWithQueryExpression() {
...
+
+function testQueryActionWithInterpolationRegexpLangLibs() {
+    string[] res = [];
+
+    from var item in ["a", "aab", "bc", "ac"] 
+    do {
+        if re `${"a.*"}`.isFullMatch(item) {
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/query/query_action_or_expr.bal:945]: This is not what I meant. Please use a variable reference and create an interpolated compound pattern within the query clause

Comment (LakshanWeerasinghe) [reply tests/jballerina-unit-test/src/test/resources/test-src/query/query_action_or_expr.bal]: Updated.

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**